src/plugins/intel_gpu/include/intel_gpu/graph/network.hpp

```diff
@@ -215,6 +216,7 @@ struct network {
     bool _is_dynamic = false;
     bool _enable_profiling = false;
     bool _reset_arguments;
+    memory::ptr _ptr;
```

Comment:
I think it's better to hold the pointer to the base buffer in the sub-buffer itself, thus their lifetimes will be better aligned